test/es-module/test-esm-resolve-type.js

```diff
@@ -98,86 +98,154 @@ try {
     }
   };
 
-  // Create a dummy dual package
-  //
-  /**
-   * this creates following directory structure:
-   *
-   * ./node_modules:
-   *   |-> my-dual-package
-   *       |-> es
-   *           |-> index.js
-   *           |-> package.json [2]
-   *       |-> lib
-   *           |-> index.js
-   *       |->package.json [1]
-   *
-   * [1] - main package.json of the package
-   *     - it contains:
-   *             - type: 'commonjs'
-   *             - main: 'lib/mainfile.js'
-   *             - conditional exports for 'require' (lib/index.js) and
-   *                                       'import' (es/index.js)
-   * [2] - package.json add-on for the import case
-   *     - it only contains:
-   *             - type: 'module'
-   *
-   * in case the package is consumed as an ESM by importing it:
-   *    import * as my-package from 'my-dual-package'
-   * it will cause the resolve method to return:
-   *  {
-   *     url: '<base_path>/node_modules/my-dual-package/es/index.js',
-   *     format: 'module'
-   *  }
-   *
-   *  following testcase ensures that resolve works correctly in this case
-   *  returning the information as specified above. Source for 'url' value
-   *  is [1], source for 'format' value is [2]
-   */
+  function testDualPackageWithJsMainScriptAndModuleType() {
+    // Create a dummy dual package
+    //
+    /**
+     * this creates following directory structure:
+     *
+     * ./node_modules:
+     *   |-> my-dual-package
+     *       |-> es
+     *           |-> index.js
+     *           |-> package.json [2]
+     *       |-> lib
+     *           |-> index.js
+     *       |->package.json [1]
+     *
+     * [1] - main package.json of the package
+     *     - it contains:
+     *             - type: 'commonjs'
+     *             - main: 'lib/mainfile.js'
+     *             - conditional exports for 'require' (lib/index.js) and
+     *                                       'import' (es/index.js)
+     * [2] - package.json add-on for the import case
+     *     - it only contains:
+     *             - type: 'module'
+     *
+     * in case the package is consumed as an ESM by importing it:
```

Comment:
```suggestion
     * in case the package is consumed as ESM:
```

Reply:
Importing a package doesn't mean the resulting module is ESM, it could be whatever.
```suggestion
     * in case the package is imported:
```

Reply:
done